### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by mchaver**
@msewell17
@smurphy8  

I added some basic tests, stack support, and an IO function to print servant errors. If you want to be able to run the tests, please check the README.md (you need to have a node package 'xmlhttprequest' in a place where GHCJS can find it and run the test server, can probably set the server up automatically with Shake)

### CODE REVIEW

File: ghcjs-servant-client.cabal [hunk 1]
```diff
@@ -1,55 +1,70 @@
...
-  <https://github.com/haskell-servant/servant/blob/master/servant-client/CHANGELOG.md CHANGELOG>
+synopsis:            undefined
+description:         undefined
 license:             BSD3
 license-file:        LICENSE
-author:              Alp Mestanogullari, Sönke Hahn, Julian K. Arni
+author:              Kevin Cotrone, Michael Sewell, James M.C. Haver II
 maintainer:          alpmestan@gmail.com
-copyright:           2014 Zalora South East Asia Pte Ltd
+copyright:           2016 Plow Technologies
```
Comment (smurphy8) [inline ghcjs-servant-client.cabal:9]: lol, yeah this is a good call

File: ghcjs-servant-client.cabal [hunk 2]
```diff
@@ -1,55 +1,70 @@
...
-  See <http://haskell-servant.github.io/tutorial/client.html the client section of the tutorial>.
-  .
-  <https://github.com/haskell-servant/servant/blob/master/servant-client/CHANGELOG.md CHANGELOG>
+synopsis:            undefined
+description:         undefined
 license:             BSD3
 license-file:        LICENSE
-author:              Alp Mestanogullari, Sönke Hahn, Julian K. Arni
+author:              Kevin Cotrone, Michael Sewell, James M.C. Haver II
 maintainer:          alpmestan@gmail.com
```
Comment (smurphy8) [inline ghcjs-servant-client.cabal]: why did yo uleave this one?

File: test-server/LICENSE [hunk 3]
```diff
@@ -0,0 +1,30 @@
+Copyright Author name here (c) 2016
```
Comment (smurphy8) [inline test-server/LICENSE:1]: lets MIT license this

**Review by smurphy8 — APPROVED** [review]
:shipit: I had a few questions about License and email but this is a cool PR!

File: ghcjs-servant-client.cabal (hunk 2 above)
Comment (mchaver) [reply ghcjs-servant-client.cabal]: oh, forgot the email, i'll fix it

**Pull request merged**